client/src/app/videos/+video-edit/video-add-components/video-upload.component.ts

```diff
@@ -146,13 +149,19 @@ export class VideoUploadComponent extends VideoSend implements OnInit, OnDestroy
       return
     }
 
+    // Build name field
     const nameWithoutExtension = videofile.name.replace(/\.[^/.]+$/, '')
     let name: string
 
     // If the name of the file is very small, keep the extension
     if (nameWithoutExtension.length < 3) name = videofile.name
     else name = nameWithoutExtension
 
+    // Force user to wait transcoding for unsupported video types in web browsers
+    if (!videofile.name.endsWith('.mp4') && !videofile.name.endsWith('.webm') && !videofile.name.endsWith('.ogv')) {
```

Comment:
rather than just testing file extensions, testing the *streamability* (yeah, new word :stuck_out_tongue:) of the file could be a good addition: https://framagit.org/rigelk/kitchensink/blob/master/src/ffmpeg-isStreamable/index.ts